### ISSUE TITLE
Downloads deleted before listened to

### ISSUE THREAD

**Issue by jhoc2000** [einmalfel/PodListen#63]
I have downloaded some podcasts with the intention of listening to them when I don't have internet access but if I wait too long the downloads are deleted before I get a chance to listen to them. It seems to take a week to be deleted.

Can this please be fixed. That is, don't delete a podcast unless it is listened to. 

I store the podcasts on the SDcard of this makes a difference.

**Comment by pevu** [comment]
Same thing here. I have a Huawei P8 lite and thought it was due to Huawei's aggressive cleanup policy, but the issue persists when I whitelist PodListen. Not sure if its a PodListen or Huawei's bug or a PodListen feature.

If it's a feature, PodListen is pretty conservative about downloading downloading podcasts already, as it doesn't download podcasts added to the playlist by default. It could be safely assumed that downloaded podcasts will be listened to. And deleting podcasts has considerable impact on how the app works, as some podcasts will have to be downloaded again and it's unclear which ones. To find out you have to try and play them and fail.

**Comment by jhoc2000** [comment]
I have experienced exactly what you mentioned so I expect that it is not a hardware problem. I guess it's up to the programmers to decide if it's the app or the operating system. I'm running Android Marshmallow 6.0.1.

**Comment by QiangF** [comment]
+1 for letting user decide when the podcast get deleted.

**Comment by dcorking** [comment]
I think that this is a duplicate of #62, in which case it is fixed on the master branch.

**Comment by jhoc2000** [comment]
Yes, this is the same issue. I have version 1.3.6 on Android 6.0.1 but the problem is not fixed so some other code needs to be written to solve the problem. 

It may be relevant that I download to the SDcard rather than the smartphone's internal memory. 

Merry Christmas.

**Comment by dcorking** [comment]
The fix for issue #62 is newer than version 1.3.6. I suggest you re-test with master branch, or wait for a new release from the author.

**Comment by jhoc2000** [comment]
According to F-Droid I have the latest version. When will a working version be available on F-Droid?

**Comment by Dazrok2** [comment]
Any update on this issue?

**Comment by jhoc2000** [comment]
I've been told that this problem is fixed in versions above 1.3.6 but I haven't been able to find a higher version even on the Google Play Store.

**Comment by ildar** [comment]
Emm, PodListen (Podcast player with lightweight interface) -
https://f-droid.org/app/com.einmalfel.podlisten ?

**Comment by dcorking** [comment]
podlisten above 1.3.6 has not been released. The fix is in master branch only. It is up to the author to release when they are ready; I don't think keeping a duplicate issue open helps them.  In the meantime, I imagine it would help if you build and test the master branch to confirm the fix for #62 works for you.

**Comment by jhoc2000** [comment]
Thank you for the follow up but I don't know how to build on an Android 6.0.1 smartphone. I don't even have root access.

**Comment by dcorking** [comment]
In that case it's probably good Github etiquette to close the issue and wait for the author to build a new release for your chosen app store. (By way of background, you can't build on an Android device, you would need to learn to use an Android SDK on a computer.)
